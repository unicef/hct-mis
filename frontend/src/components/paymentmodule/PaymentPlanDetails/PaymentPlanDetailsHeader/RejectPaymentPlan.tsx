import {
  Box,
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
} from '@material-ui/core';
import * as Yup from 'yup';
import { Field, Form, Formik } from 'formik';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { DialogContainer } from '../../../../containers/dialogs/DialogContainer';
import { DialogFooter } from '../../../../containers/dialogs/DialogFooter';
import { DialogTitleWrapper } from '../../../../containers/dialogs/DialogTitleWrapper';
import { useSnackbar } from '../../../../hooks/useSnackBar';
import { FormikTextField } from '../../../../shared/Formik/FormikTextField/FormikTextField';
import { LoadingButton } from '../../../core/LoadingButton';
import { GreyText } from '../../../core/GreyText';
import { ErrorButton } from '../../../core/ErrorButton';

export interface RejectPaymentPlanProps {
  paymentPlanId: string;
}

export const RejectPaymentPlan = ({
  paymentPlanId,
}: RejectPaymentPlanProps): React.ReactElement => {
  const { t } = useTranslation();
  const [rejectDialogOpen, setRejectDialogOpen] = useState(false);

  const { showMessage } = useSnackbar();
  // const [mutate] = useActivateCashPlanPaymentVerificationMutation();
  // const activate = async (): Promise<void> => {
  //   try {
  //     await mutate({
  //       variables: { cashPlanVerificationId },
  //       refetchQueries,
  //     });
  //   } catch (error) {
  //     /* eslint-disable-next-line no-console */
  //     console.log('error', error?.graphQLErrors);
  //     if (
  //       error?.graphQLErrors?.[0]?.validationErrors
  //         ?.activateCashPlanPaymentVerification?.phone_numbers
  //     ) {
  //       showMessage(
  //         error?.graphQLErrors?.[0]?.validationErrors?.activateCashPlanPaymentVerification?.phone_numbers.join(
  //           '\n',
  //         ),
  //       );
  //     } else {
  //       showMessage(t('Error during activating.'));
  //     }
  //   }

  //   showMessage(t('Verification plan has been activated.'));
  // };
  const initialValues = {
    comment: '',
  };

  const validationSchema = Yup.object().shape({
    comment: Yup.string()
      .min(2, 'Too short')
      .max(255, 'Too long'),
  });

  return (
    <>
      <Formik
        initialValues={initialValues}
        onSubmit={(values, { resetForm }) => {
          console.log('Reject');
          resetForm({});
        }}
        validationSchema={validationSchema}
      >
        {({ submitForm }) => (
          <>
            <Box p={2}>
              <ErrorButton onClick={() => setRejectDialogOpen(true)}>
                {t('Reject')}
              </ErrorButton>
            </Box>
            <Dialog
              open={rejectDialogOpen}
              onClose={() => setRejectDialogOpen(false)}
              scroll='paper'
              aria-labelledby='form-dialog-title'
              maxWidth='md'
            >
              <DialogTitleWrapper>
                <DialogTitle id='scroll-dialog-title'>
                  {t('Reject')}
                </DialogTitle>
              </DialogTitleWrapper>
              <DialogContent>
                <DialogContainer>
                  <Box p={5}>
                    {t('Are you sure you want to reject this payment plan?')}
                  </Box>
                  <Box p={5}>
                    <GreyText>
                      {t(
                        'Note: Upon proceeding this payment plan will be automatically moved to locked status.',
                      )}
                    </GreyText>
                  </Box>
                  <Form>
                    <Field
                      name='comment'
                      multiline
                      fullWidth
                      variant='filled'
                      label='Comment (optional)'
                      component={FormikTextField}
                    />
                  </Form>
                </DialogContainer>
              </DialogContent>
              <DialogFooter>
                <DialogActions>
                  <Button onClick={() => setRejectDialogOpen(false)}>
                    CANCEL
                  </Button>
                  <LoadingButton
                    loading={true}
                    type='submit'
                    color='primary'
                    variant='contained'
                    onClick={() => console.log(paymentPlanId)}
                    data-cy='button-submit'
                  >
                    {t('Reject')}
                  </LoadingButton>
                </DialogActions>
              </DialogFooter>
            </Dialog>
          </>
        )}
      </Formik>
    </>
  );
};
