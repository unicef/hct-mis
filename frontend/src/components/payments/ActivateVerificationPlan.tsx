import { Box, Button, DialogContent, DialogTitle } from '@material-ui/core';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { Dialog } from '../../containers/dialogs/Dialog';
import { DialogActions } from '../../containers/dialogs/DialogActions';
import { DialogContainer } from '../../containers/dialogs/DialogContainer';
import { DialogFooter } from '../../containers/dialogs/DialogFooter';
import { DialogTitleWrapper } from '../../containers/dialogs/DialogTitleWrapper';
import { usePaymentRefetchQueries } from '../../hooks/usePaymentRefetchQueries';
import { useSnackbar } from '../../hooks/useSnackBar';
import { useActivatePaymentVerificationPlanMutation } from '../../__generated__/graphql';

export interface ActivateVerificationPlanProps {
  paymentVerificationPlanId: string;
  cashOrPaymentPlanId: string;
}

export function ActivateVerificationPlan({
  paymentVerificationPlanId,
  cashOrPaymentPlanId,
}: ActivateVerificationPlanProps): React.ReactElement {
  const refetchQueries = usePaymentRefetchQueries(cashOrPaymentPlanId);
  const { t } = useTranslation();
  const [activateDialogOpen, setActivateDialogOpen] = useState(false);

  const { showMessage } = useSnackbar();
  const [mutate] = useActivatePaymentVerificationPlanMutation();
  const activate = async (): Promise<void> => {
    try {
      await mutate({
        variables: { paymentVerificationPlanId },
        refetchQueries,
      });
    } catch (error) {
      /* eslint-disable-next-line no-console */
      console.log('error', error?.graphQLErrors);
      if (
        error?.graphQLErrors?.[0]?.validationErrors
          ?.activateCashPlanPaymentVerification?.phone_numbers
      ) {
        showMessage(
          error?.graphQLErrors?.[0]?.validationErrors?.activateCashPlanPaymentVerification?.phone_numbers.join(
            '\n',
          ),
        );
      } else {
        showMessage(t('Error during activating.'));
      }
    }

    showMessage(t('Verification plan has been activated.'));
  };
  return (
    <>
      <Box p={2}>
        <Button
          color='primary'
          variant='contained'
          onClick={() => setActivateDialogOpen(true)}
          data-cy='button-activate-plan'
        >
          {t('ACTIVATE')}
        </Button>
      </Box>
      <Dialog
        open={activateDialogOpen}
        onClose={() => setActivateDialogOpen(false)}
        scroll='paper'
        aria-labelledby='form-dialog-title'
        maxWidth='md'
      >
        <DialogTitleWrapper>
          <DialogTitle id='scroll-dialog-title'>
            {t('Activate Verification Plan')}
          </DialogTitle>
        </DialogTitleWrapper>
        <DialogContent>
          <DialogContainer>
            <Box p={5}>
              {t(
                'Are you sure you want to activate the Verification Plan for this Cash Plan?',
              )}
            </Box>
          </DialogContainer>
        </DialogContent>
        <DialogFooter>
          <DialogActions>
            <Button onClick={() => setActivateDialogOpen(false)}>CANCEL</Button>
            <Button
              type='submit'
              color='primary'
              variant='contained'
              onClick={() => activate()}
              data-cy='button-submit'
            >
              {t('ACTIVATE')}
            </Button>
          </DialogActions>
        </DialogFooter>
      </Dialog>
    </>
  );
}
