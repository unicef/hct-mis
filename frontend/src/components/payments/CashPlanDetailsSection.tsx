import { Box, Grid, Typography } from '@material-ui/core';
import React from 'react';
import { Doughnut } from 'react-chartjs-2';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import { useBusinessArea } from '../../hooks/useBusinessArea';
import { countPercentage } from '../../utils/utils';
import { CashPlanQuery } from '../../__generated__/graphql';
import { BlackLink } from '../core/BlackLink';
import { LabelizedField } from '../core/LabelizedField';
import { UniversalMoment } from '../core/UniversalMoment';

const Title = styled.div`
  padding-bottom: ${({ theme }) => theme.spacing(8)}px;
`;

const ChartContainer = styled.div`
  width: 150px;
  height: 150px;
`;

const BorderLeftBox = styled.div`
  padding-left: ${({ theme }) => theme.spacing(6)}px;
  border-left: 1px solid #e0e0e0;
  height: 100%;
`;

interface CashPlanDetailsSectionProps {
  cashPlan: CashPlanQuery['cashPlan'];
}

export const CashPlanDetailsSection = ({
  cashPlan,
}: CashPlanDetailsSectionProps): React.ReactElement => {
  const { t } = useTranslation();
  const businessArea = useBusinessArea();

  const bankReconciliationSuccessPercentage = countPercentage(
    cashPlan.bankReconciliationSuccess,
    cashPlan.paymentRecords.totalCount,
  );

  const bankReconciliationErrorPercentage = countPercentage(
    cashPlan.bankReconciliationError,
    cashPlan.paymentRecords.totalCount,
  );

  return (
    <Grid container>
      <Grid item xs={9}>
        <Title>
          <Typography variant='h6'>{t('Cash Plan Details')}</Typography>
        </Title>
        <Grid container>
          {[
            { label: t('PROGRAMME NAME'), value: cashPlan.program.name },
            {
              label: t('PROGRAMME ID'),
              value: (
                <BlackLink
                  to={`/${businessArea}/programs/${cashPlan.program.id}`}
                >
                  {cashPlan.program?.caId}
                </BlackLink>
              ),
            },
            {
              label: t('PAYMENT RECORDS'),
              value: cashPlan.availablePaymentRecordsCount,
            },
            {
              label: t('START DATE'),
              value: <UniversalMoment>{cashPlan.startDate}</UniversalMoment>,
            },
            {
              label: t('END DATE'),
              value: <UniversalMoment>{cashPlan.endDate}</UniversalMoment>,
            },
          ].map((el) => (
            <Grid item xs={4} key={el.label}>
              <Box pt={2} pb={2}>
                <LabelizedField label={el.label}>{el.value}</LabelizedField>
              </Box>
            </Grid>
          ))}
        </Grid>
      </Grid>
      <Grid item xs={3}>
        <BorderLeftBox>
          <Title>
            <Typography variant='h6'>{t('Bank reconciliation')}</Typography>
          </Title>
          <Grid container>
            <Grid item xs={6}>
              <Grid container direction='column'>
                <LabelizedField label={t('SUCCESSFUL')}>
                  <p>{bankReconciliationSuccessPercentage}%</p>
                </LabelizedField>
                <LabelizedField label={t('ERRONEOUS')}>
                  <p>{bankReconciliationErrorPercentage}%</p>
                </LabelizedField>
              </Grid>
            </Grid>
            <Grid item xs={6}>
              <ChartContainer>
                <Doughnut
                  width={200}
                  height={200}
                  options={{
                    maintainAspectRatio: false,
                    cutoutPercentage: 80,
                    legend: {
                      display: false,
                    },
                  }}
                  data={{
                    labels: [t('SUCCESSFUL'), t('ERRONEOUS')],
                    datasets: [
                      {
                        data: [
                          bankReconciliationSuccessPercentage,
                          bankReconciliationErrorPercentage,
                        ],
                        backgroundColor: ['#00509F', '#FFAA1F'],
                        hoverBackgroundColor: ['#00509F', '#FFAA1F'],
                      },
                    ],
                  }}
                />
              </ChartContainer>
            </Grid>
          </Grid>
        </BorderLeftBox>
      </Grid>
    </Grid>
  );
};
