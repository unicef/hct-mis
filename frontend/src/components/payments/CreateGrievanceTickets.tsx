import React, { useState } from 'react';
import { Button, DialogContent, DialogTitle, Box } from '@material-ui/core';
import styled from 'styled-components';
import { Dialog } from '../../containers/dialogs/Dialog';
import { DialogActions } from '../../containers/dialogs/DialogActions';
import { useSnackbar } from '../../hooks/useSnackBar';

export interface Props {
  disabled: boolean;
  grievanceTicketsNumber: number;
}
const DialogTitleWrapper = styled.div`
  border-bottom: 1px solid ${({ theme }) => theme.hctPalette.lighterGray};
`;

const DialogFooter = styled.div`
  padding: 12px 16px;
  margin: 0;
  border-top: 1px solid ${({ theme }) => theme.hctPalette.lighterGray};
  text-align: right;
`;

const DialogContainer = styled.div`
  width: 700px;
`;

export function CreateGrievanceTickets({
  disabled,
  grievanceTicketsNumber,
}: Props): React.ReactElement {
  const [DialogOpen, setDialogOpen] = useState(false);

  const createGrievanceTickets = (): void => {
    setDialogOpen(false);
  };
  return (
    <>
      <Button
        color='primary'
        disabled={disabled}
        onClick={() => setDialogOpen(true)}
        data-cy='button-ed-plan'
      >
        CREATE GRIEVANCE TICKETS
      </Button>
      <Dialog
        open={DialogOpen}
        onClose={() => setDialogOpen(false)}
        scroll='paper'
        aria-labelledby='form-dialog-title'
      >
        <DialogTitleWrapper>
          <DialogTitle id='scroll-dialog-title'>
            Create Grievance Tickets
          </DialogTitle>
        </DialogTitleWrapper>
        <DialogContent>
          <DialogContainer>
            <Box p={5}>
              <div>
                You will generate {grievanceTicketsNumber} grievance ticket
                {grievanceTicketsNumber === 1 ? null : 's'}. Are you sure you
                want to continue?
              </div>
            </Box>
          </DialogContainer>
        </DialogContent>
        <DialogFooter>
          <DialogActions>
            <Button onClick={() => setDialogOpen(false)}>CANCEL</Button>
            <Button
              type='submit'
              color='primary'
              variant='contained'
              onClick={() => createGrievanceTickets()}
              data-cy='button-submit'
            >
              Continue
            </Button>
          </DialogActions>
        </DialogFooter>
      </Dialog>
    </>
  );
}
