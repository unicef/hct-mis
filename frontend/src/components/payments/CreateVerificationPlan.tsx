import {
  Box,
  Button,
  DialogContent,
  DialogTitle,
  Grid,
  Tab,
  Tabs,
  Typography,
} from '@material-ui/core';
import { Field, Form, Formik } from 'formik';
import React, { useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import { Dialog } from '../../containers/dialogs/Dialog';
import { DialogActions } from '../../containers/dialogs/DialogActions';
import { useBusinessArea } from '../../hooks/useBusinessArea';
import { usePaymentRefetchQueries } from '../../hooks/usePaymentRefetchQueries';
import { useSnackbar } from '../../hooks/useSnackBar';
import { FormikCheckboxField } from '../../shared/Formik/FormikCheckboxField';
import { FormikMultiSelectField } from '../../shared/Formik/FormikMultiSelectField';
import { FormikRadioGroup } from '../../shared/Formik/FormikRadioGroup';
import { FormikSelectField } from '../../shared/Formik/FormikSelectField';
import { FormikSliderField } from '../../shared/Formik/FormikSliderField';
import { FormikTextField } from '../../shared/Formik/FormikTextField';
import { getPercentage } from '../../utils/utils';
import {
  useAllAdminAreasQuery,
  useAllRapidProFlowsQuery,
  useCreateCashPlanPaymentVerificationMutation,
  useSampleSizeLazyQuery,
} from '../../__generated__/graphql';
import { ButtonTooltip } from '../core/ButtonTooltip';
import { FormikEffect } from '../core/FormikEffect';
import { TabPanel } from '../core/TabPanel';

const DialogTitleWrapper = styled.div`
  border-bottom: 1px solid ${({ theme }) => theme.hctPalette.lighterGray};
`;

const DialogFooter = styled.div`
  padding: 12px 16px;
  margin: 0;
  border-top: 1px solid ${({ theme }) => theme.hctPalette.lighterGray};
  text-align: right;
`;

const StyledTabs = styled(Tabs)`
  && {
    max-width: 500px;
  }
`;
const TabsContainer = styled.div`
  border-bottom: 1px solid #e8e8e8;
`;
const DialogContainer = styled.div`
  width: 700px;
`;

const initialValues = {
  confidenceInterval: 95,
  marginOfError: 5,
  filterAgeMin: null,
  filterAgeMax: null,
  filterSex: '',
  excludedAdminAreasFull: [],
  excludedAdminAreasRandom: [],
  verificationChannel: 'MANUAL',
  rapidProFlow: '',
  adminCheckbox: false,
  ageCheckbox: false,
  sexCheckbox: false,
};

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
function prepareVariables(cashPlanId, selectedTab, values, businessArea) {
  const variables = {
    input: {
      cashPlanId,
      sampling: selectedTab === 0 ? 'FULL_LIST' : 'RANDOM',
      fullListArguments:
        selectedTab === 0
          ? {
              excludedAdminAreas: values.excludedAdminAreasFull || [],
            }
          : null,
      verificationChannel: values.verificationChannel,
      rapidProArguments:
        values.verificationChannel === 'RAPIDPRO'
          ? {
              flowId: values.rapidProFlow,
            }
          : null,
      randomSamplingArguments:
        selectedTab === 1
          ? {
              confidenceInterval: values.confidenceInterval * 0.01,
              marginOfError: values.marginOfError * 0.01,
              excludedAdminAreas: values.adminCheckbox
                ? values.excludedAdminAreasRandom
                : [],
              age: values.ageCheckbox
                ? { min: values.filterAgeMin, max: values.filterAgeMax }
                : null,
              sex: values.sexCheckbox ? values.filterSex : null,
            }
          : null,
      businessAreaSlug: businessArea,
    },
  };
  return variables;
}

export interface Props {
  cashPlanId: string;
  disabled: boolean;
  canCreatePaymentVerificationPlan: boolean;
}
export function CreateVerificationPlan({
  cashPlanId,
  disabled,
  canCreatePaymentVerificationPlan,
}: Props): React.ReactElement {
  const refetchQueries = usePaymentRefetchQueries(cashPlanId);
  const { t } = useTranslation();
  const [open, setOpen] = useState(false);
  const [selectedTab, setSelectedTab] = useState(0);
  const { showMessage } = useSnackbar();
  const [mutate] = useCreateCashPlanPaymentVerificationMutation();
  const businessArea = useBusinessArea();
  const [formValues, setFormValues] = useState(initialValues);

  const { data: rapidProFlows } = useAllRapidProFlowsQuery({
    variables: {
      businessAreaSlug: businessArea,
    },
  });
  const { data } = useAllAdminAreasQuery({
    variables: {
      first: 100,
      businessArea,
    },
  });

  const [loadSampleSize, { data: sampleSizesData }] = useSampleSizeLazyQuery({
    variables: prepareVariables(
      cashPlanId,
      selectedTab,
      formValues,
      businessArea,
    ),
    fetchPolicy: 'network-only',
  });

  useEffect(() => {
    loadSampleSize();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [formValues, open]);

  const submit = async (values): Promise<void> => {
    const { errors } = await mutate({
      variables: prepareVariables(
        cashPlanId,
        selectedTab,
        values,
        businessArea,
      ),
      refetchQueries,
    });
    setOpen(false);

    if (errors) {
      showMessage(t('Error while submitting'));
      return;
    }
    showMessage(t('New verification plan created.'));
  };

  const mappedAdminAreas = data?.allAdminAreas?.edges?.length
    ? data.allAdminAreas.edges.map((el) => ({
        value: el.node.id,
        name: el.node.title,
      }))
    : [];

  const handleFormChange = (values): void => {
    setFormValues(values);
  };

  const getSampleSizePercentage = (): string => {
    return `(${getPercentage(
      sampleSizesData?.sampleSize?.sampleSize,
      sampleSizesData?.sampleSize?.paymentRecordCount,
    )})`;
  };

  const handleOpen = (): void => {
    if (canCreatePaymentVerificationPlan) {
      setOpen(true);
    } else {
      showMessage(
        t(
          'There are no payment records that could be assigned to a new verification plan.',
        ),
      );
    }
  };

  return (
    <Formik initialValues={initialValues} onSubmit={submit}>
      {({ submitForm, values, setValues }) => (
        <Form>
          <FormikEffect
            values={values}
            onChange={() => handleFormChange(values)}
          />
          <Box mr={2}>
            <ButtonTooltip
              disabled={disabled}
              color='primary'
              variant='contained'
              onClick={() => handleOpen()}
              data-cy='button-new-plan'
            >
              {t('CREATE VERIFICATION PLAN')}
            </ButtonTooltip>
          </Box>
          <Dialog
            open={open}
            onClose={() => setOpen(false)}
            scroll='paper'
            aria-labelledby='form-dialog-title'
            maxWidth='md'
          >
            <DialogTitleWrapper>
              <DialogTitle id='scroll-dialog-title'>
                {t('Create Verification Plan')}
              </DialogTitle>
            </DialogTitleWrapper>
            <DialogContent>
              <DialogContainer>
                <TabsContainer>
                  <StyledTabs
                    value={selectedTab}
                    onChange={(
                      event: React.ChangeEvent<{}>,
                      newValue: number,
                    ) => {
                      setValues(initialValues);
                      setFormValues(initialValues);
                      setSelectedTab(newValue);
                    }}
                    indicatorColor='primary'
                    textColor='primary'
                    variant='fullWidth'
                    aria-label='full width tabs example'
                  >
                    <Tab label={t('FULL LIST')} />
                    <Tab label={t('RANDOM SAMPLING')} />
                  </StyledTabs>
                </TabsContainer>
                <TabPanel value={selectedTab} index={0}>
                  {mappedAdminAreas && (
                    <Field
                      name='excludedAdminAreasFull'
                      choices={mappedAdminAreas}
                      variant='outlined'
                      label={t('Filter Out Administrative Level Areas')}
                      component={FormikMultiSelectField}
                    />
                  )}
                  <Box pt={3}>
                    <Box
                      pb={3}
                      pt={3}
                      fontSize={16}
                      fontWeight='fontWeightBold'
                    >
                      Sample size: {sampleSizesData?.sampleSize?.sampleSize} out
                      of {sampleSizesData?.sampleSize?.paymentRecordCount}{' '}
                      {getSampleSizePercentage()}
                    </Box>
                    <Box fontSize={12} color='#797979'>
                      {t('This option is recommended for RapidPro')}
                    </Box>
                    <Field
                      name='verificationChannel'
                      label={t('Verification Channel')}
                      style={{ flexDirection: 'row' }}
                      choices={[
                        { value: 'RAPIDPRO', name: 'RAPIDPRO' },
                        { value: 'XLSX', name: 'XLSX' },
                        { value: 'MANUAL', name: 'MANUAL' },
                      ]}
                      component={FormikRadioGroup}
                    />
                    {values.verificationChannel === 'RAPIDPRO' && (
                      <Field
                        name='rapidProFlow'
                        label={t('RapidPro Flow')}
                        style={{ width: '90%' }}
                        choices={
                          rapidProFlows
                            ? rapidProFlows.allRapidProFlows.map((flow) => ({
                                value: flow.id,
                                name: flow.name,
                              }))
                            : []
                        }
                        component={FormikSelectField}
                      />
                    )}
                  </Box>
                </TabPanel>
                <TabPanel value={selectedTab} index={1}>
                  <Box pt={3}>
                    <Field
                      name='confidenceInterval'
                      label={t('Confidence Interval')}
                      min={90}
                      max={99}
                      component={FormikSliderField}
                      suffix='%'
                    />
                    <Field
                      name='marginOfError'
                      label={t('Margin of Error')}
                      min={0}
                      max={9}
                      component={FormikSliderField}
                      suffix='%'
                    />
                    <Typography variant='caption'>
                      {t('Cluster Filters')}
                    </Typography>
                    <Box flexDirection='column' display='flex'>
                      <Box display='flex'>
                        <Field
                          name='adminCheckbox'
                          label={t('Administrative Level')}
                          component={FormikCheckboxField}
                        />
                        <Field
                          name='ageCheckbox'
                          label={t('Age of HoH')}
                          component={FormikCheckboxField}
                        />
                        <Field
                          name='sexCheckbox'
                          label={t('Gender of HoH')}
                          component={FormikCheckboxField}
                        />
                      </Box>
                      {values.adminCheckbox && (
                        <Field
                          name='excludedAdminAreasRandom'
                          choices={mappedAdminAreas}
                          variant='outlined'
                          label={t('Filter Out Administrative Level Areas')}
                          component={FormikMultiSelectField}
                        />
                      )}

                      <Grid container>
                        {values.ageCheckbox && (
                          <Grid item xs={12}>
                            <Grid container>
                              <Grid item xs={4}>
                                <Field
                                  name='filterAgeMin'
                                  label={t('Minimum Age')}
                                  type='number'
                                  color='primary'
                                  component={FormikTextField}
                                />
                              </Grid>
                              <Grid item xs={4}>
                                <Field
                                  name='filterAgeMax'
                                  label={t('Maximum Age')}
                                  type='number'
                                  color='primary'
                                  component={FormikTextField}
                                />
                              </Grid>
                            </Grid>
                          </Grid>
                        )}
                        {values.sexCheckbox && (
                          <Grid item xs={5}>
                            <Field
                              name='filterSex'
                              label={t('Gender')}
                              color='primary'
                              choices={[
                                { value: 'FEMALE', name: t('Female') },
                                { value: 'MALE', name: t('Male') },
                              ]}
                              component={FormikSelectField}
                            />
                          </Grid>
                        )}
                      </Grid>
                    </Box>

                    <Box
                      pb={3}
                      pt={3}
                      fontSize={16}
                      fontWeight='fontWeightBold'
                    >
                      Sample size: {sampleSizesData?.sampleSize?.sampleSize} out
                      of {sampleSizesData?.sampleSize?.paymentRecordCount}
                      {getSampleSizePercentage()}
                    </Box>
                    <Field
                      name='verificationChannel'
                      label={t('Verification Channel')}
                      style={{ flexDirection: 'row' }}
                      choices={[
                        { value: 'RAPIDPRO', name: 'RAPIDPRO' },
                        { value: 'XLSX', name: 'XLSX' },
                        { value: 'MANUAL', name: 'MANUAL' },
                      ]}
                      component={FormikRadioGroup}
                    />
                    {values.verificationChannel === 'RAPIDPRO' && (
                      <Field
                        name='rapidProFlow'
                        label='RapidPro Flow'
                        style={{ width: '90%' }}
                        choices={
                          rapidProFlows ? rapidProFlows.allRapidProFlows : []
                        }
                        component={FormikSelectField}
                      />
                    )}
                  </Box>
                </TabPanel>
              </DialogContainer>
            </DialogContent>
            <DialogFooter>
              <DialogActions>
                <Button onClick={() => setOpen(false)}>CANCEL</Button>
                <Button
                  type='submit'
                  color='primary'
                  variant='contained'
                  onClick={submitForm}
                  data-cy='button-submit'
                >
                  SAVE
                </Button>
              </DialogActions>
            </DialogFooter>
          </Dialog>
        </Form>
      )}
    </Formik>
  );
}
