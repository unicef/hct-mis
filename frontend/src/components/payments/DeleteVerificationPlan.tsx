import { Box, Button, DialogContent, DialogTitle } from '@material-ui/core';
import ClearIcon from '@material-ui/icons/Clear';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { Dialog } from '../../containers/dialogs/Dialog';
import { DialogActions } from '../../containers/dialogs/DialogActions';
import { DialogContainer } from '../../containers/dialogs/DialogContainer';
import { DialogFooter } from '../../containers/dialogs/DialogFooter';
import { DialogTitleWrapper } from '../../containers/dialogs/DialogTitleWrapper';
import { usePaymentRefetchQueries } from '../../hooks/usePaymentRefetchQueries';
import { useSnackbar } from '../../hooks/useSnackBar';
import { useDeletePaymentVerificationPlanMutation } from '../../__generated__/graphql';
import { ErrorButton } from '../core/ErrorButton';
import { ErrorButtonContained } from '../core/ErrorButtonContained';

export interface DeleteVerificationPlanProps {
  paymentVerificationPlanId: string;
  cashPlanId: string;
}

export function DeleteVerificationPlan({
  paymentVerificationPlanId,
  cashPlanId,
}: DeleteVerificationPlanProps): React.ReactElement {
  const refetchQueries = usePaymentRefetchQueries(cashPlanId);
  const { t } = useTranslation();
  const [finishDialogOpen, setFinishDialogOpen] = useState(false);
  const { showMessage } = useSnackbar();
  const [mutate] = useDeletePaymentVerificationPlanMutation();

  const handleDeleteVerificationPlan = async (): Promise<void> => {
    const { errors } = await mutate({
      variables: { paymentVerificationPlanId },
      refetchQueries,
    });
    if (errors) {
      showMessage(t('Error while submitting'));
      return;
    }
    showMessage(t('Verification plan has been deleted.'));
  };
  return (
    <>
      <Box p={2}>
        <ErrorButton
          startIcon={<ClearIcon />}
          onClick={() => setFinishDialogOpen(true)}
          data-cy='button-delete-plan'
        >
          DELETE
        </ErrorButton>
      </Box>
      <Dialog
        open={finishDialogOpen}
        onClose={() => setFinishDialogOpen(false)}
        scroll='paper'
        aria-labelledby='form-dialog-title'
        maxWidth='md'
      >
        <DialogTitleWrapper>
          <DialogTitle id='scroll-dialog-title'>
            {t('Delete Verification Plan')}
          </DialogTitle>
        </DialogTitleWrapper>
        <DialogContent>
          <DialogContainer>
            <Box p={5}>
              <div>
                {t(
                  'Are you sure you would like to delete this verification plan?',
                )}
              </div>
            </Box>
          </DialogContainer>
        </DialogContent>
        <DialogFooter>
          <DialogActions>
            <Button onClick={() => setFinishDialogOpen(false)}>
              {t('CANCEL')}
            </Button>
            <ErrorButtonContained
              type='submit'
              onClick={() => handleDeleteVerificationPlan()}
              data-cy='button-submit'
            >
              {t('DELETE')}
            </ErrorButtonContained>
          </DialogActions>
        </DialogFooter>
      </Dialog>
    </>
  );
}
