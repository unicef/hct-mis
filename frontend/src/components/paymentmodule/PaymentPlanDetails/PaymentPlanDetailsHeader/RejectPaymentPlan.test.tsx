import { MockedProvider } from '@apollo/react-testing';
import React from 'react';
import { act } from 'react-dom/test-utils';
import wait from 'waait';
import { fakeApolloPaymentPlan } from '../../../../../fixtures/payments/fakeApolloPaymentPlan';
import { fakeActionPpMutation } from '../../../../../fixtures/payments/fakeApolloActionPaymentPlanMutation';
import { render } from '../../../../testUtils/testUtils';
import { RejectPaymentPlan } from './RejectPaymentPlan';

describe(
  'components/paymentmodule/PaymentPlanDetails/PaymentPlanDetailsHeader/RejectPaymentPlan',
  () => {
    it('should render', async () => {
      const { container } = render(
        <MockedProvider
          addTypename={false}
          mocks={fakeActionPpMutation}
        >
          <RejectPaymentPlan paymentPlanId={fakeApolloPaymentPlan.id} />
        </MockedProvider>
        ,
      );

      await act(() => wait(0)); // wait for the mutation to complete

      expect(container).toMatchSnapshot();
    });
  });
