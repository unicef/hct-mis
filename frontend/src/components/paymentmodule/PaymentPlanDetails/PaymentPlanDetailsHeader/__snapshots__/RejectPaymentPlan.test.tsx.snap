// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/paymentmodule/PaymentPlanDetails/PaymentPlanDetailsHeader/RejectPaymentPlan should render 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-error-2"
      data-cy="button-reject-payment-plan"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Reject
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
